### ISSUE TITLE
No defaulting rules for polymorphic types

### ISSUE THREAD

**Issue by Shimuuar** [theam/haskell-do#94]
1. Try to evaluate block `10` 
2. It will fail with error about ambiguous type `a0`.

Reason for that is clear. GHC needs to select `AskInliterate` instance for literal 10 but have no way to do so. I think this is something that couldn't be solved without compiler help.

**Comment by NickSeagull** [comment]
Yes, the `eval` block should be `show 10`, or rather it won't work
